runtime/master/src/main/java/edu/snu/nemo/runtime/master/BlockMetadata.java

```diff
@@ -108,9 +97,9 @@ BlockState getBlockState() {
   }
 
   /**
-   * @return the future of the location of this Block.
+   * @return the handler of block location requests.
    */
-  synchronized CompletableFuture<String> getLocationFuture() {
-    return locationFuture;
+  synchronized BlockManagerMaster.BlockLocationRequestHandler getLocationHandler() {
```

Comment:
Remove `synchronized`? All of the callers of this method acquires a lock.

Reply:
They only acquire a "read lock", but this method have to be synchronized with the block state change.

Reply:
Got it. I'll give it another look.